### PR TITLE
Fixed #94 for AMQP Identity Promotion

### PR DESCRIPTION
**Pull request by vjwang**
- Changed the package names in the Promote User Identity into the AMQP Protocol topic
- Added a migration step for those using the old package names in 4.0 
- Added an x-ref from the Upgrade checklist to point to the migration step
- Fixed a link to the Glossary

### CODE REVIEW

**Comment by vjwang** [comment]
@krismcqueen Please review the Promote User Identity into the AMQP Protocol (p_auth_user_identity_promotion.md) and Upgrade the Gateway (o_upgrade.md) topics. See the commit message for the changes.

File: doc/security/p_auth_user_identity_promotion.md [hunk 1]
```diff
@@ -69,7 +69,7 @@ The following procedure describes how to implement AMQP user identity promotion
 
 2. Compile your LoginModule and include it in a JAR file that you put into the `GATEWAY_HOME/lib` directory.
 
-  **Note:** These instructions assume the jar `com.kaazing.gateway.amqp.server.spi.jar` is added to the compile-time classpath of the login module.
+  **Note:** These instructions assume the jar `com.kaazing.gateway.service.amqp.spi.AmqpPrincipal.jar` is added to the compile-time classpath of the login module.
```
Comment (krismcqueen) [inline doc/security/p_auth_user_identity_promotion.md]: enterprise.gateway.amqp.spi.jar is the JAR file with the AmqpPrincipal class defined.

**Comment by krismcqueen** [comment]
@vjwang Looks great, just need to fix one JAR file name.

**Comment by vjwang** [comment]
@krismcqueen Updated. Thanks!

**Pull request merged**